app/workers/projects/reorder_children_job.rb

```diff
@@ -0,0 +1,71 @@
+#-- encoding: UTF-8
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2021 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Projects
+  class ReorderChildrenJob < ApplicationJob
+    def perform
+      Rails.logger.info { "Resorting siblings by name in the project's nested set." }
+      Project.transaction { reorder! }
+    end
+
+    private
+
+    def reorder!
+      Project
+        .where(id: unique_parent_ids)
+        .find_each(&method(:reorder_children))
+    end
+
+    def unique_parent_ids
+      Project
+        .where.not(parent_id: nil)
+        .select(:parent_id)
+        .distinct
+    end
+
+    def reorder_children(parent)
```

Comment:
Wouldn't a `Project.rebuild!` have the same effect since we have configured the sorting column in `acts_as_nested_set`?